### ISSUE TITLE
Actual C_Cpp.default.configurationProvider randomly changes

### ISSUE THREAD

**Issue by myocytebd** [microsoft/vscode-cpptools#13038]
### Environment

- OS and Version: Linux x64 6.8.0-49-generic (Ubuntu 22.04.5 LTS)
- VS Code Version: 1.95.3
- C/C++ Extension Version: v1.22.11
- ms-vscode.makefile-tools: v0.11.13
- mesonbuild.mesonbuild: v1.26.0

### Bug Summary and Steps to Reproduce

Bug Summary:  
Unexpected `C_Cpp.default.configurationProvider` modification appear in `C/C++: Log Diagnostics`, although setting is not modified.
As a result, some files' intellisense is incorrect.

1. My project happens to have a broken meson.build file and valid autotools/makefile.
2. Ensure that `C_Cpp.default.configurationProvider` is always `ms-vscode.makefile-tools`.
3. Use vscode and run make/configure.
4. After some time, intellisense of some (newly opened or created) files went wrong.
  Check `C/C++: Log Diagnostics` and find out that compiler/flags are wrong.
  Further check `C/C++: Log Diagnostics` and find this:
```
Modified Settings:
{
    "C_Cpp.default.configurationProvider": "mesonbuild.mesonbuild"
}
```
5. However, this setting was never changed.
  Check settings and command palette: setting is `ms-vscode.makefile-tools`.
6. In command palette: set configurationProvider to `None` then back to `ms-vscode.makefile-tools`. And the problem seems temporarily fixed:
  Intellisense is back to normal, and `mesonbuild.mesonbuild` is gone from `C/C++: Log Diagnostics`.
7. Change makefile-tools version (and restart extension), or reload windows, also have a chance to trigger the unexpected `Modified Settings`.

### Configuration and Logs

```shell
`C/C++: Log Diagnostics` is not persistent.  
I don't know the exact condition to reproduce this issue (or if it is random).
```

### Additional context

Expected Behavior:  
- If for some reason, `ms-vscode.makefile-tools` fails, actual `C_Cpp.default.configurationProvider` should fallback to `None` instead of an unpicked provider.
- If actual `C_Cpp.default.configurationProvider` changed, an error/warning message should be presented to user.
- `ms-vscode.makefile-tools` should not fail.

**Comment by bobbrow** [comment]
It appears that the Meson Build extension has code that modifies the configuration provider setting.  I see that that extension has a setting to disable this behavior that you might want to try: `"mesonbuild.modifySettings"`

https://github.com/mesonbuild/vscode-meson/blob/869daf426b85d671aac5f52e423ed6e04c0029bb/package.json#L261

**Comment by myocytebd** [comment]
> It appears that the Meson Build extension has code that modifies the configuration provider setting. I see that that extension has a setting to disable this behavior that you might want to try: `"mesonbuild.modifySettings"`
> 
> https://github.com/mesonbuild/vscode-meson/blob/869daf426b85d671aac5f52e423ed6e04c0029bb/package.json#L261

Thanks.
However the main reason that I opened the issue as bug was because of:
```
5. However, this setting was never changed.
   Check settings and command palette: setting is ms-vscode.makefile-tools.
   ^^
```
vscode UI apparently desynced with actual provider settings.

**Comment by bobbrow** [comment]
I don't think we can guarantee stable behavior when another extension is writing settings.  Does disabling that setting fix the problem for you?  Did you set the provider to Makefile Tools in the User, Workspace, or Folder settings tab?
